bin/resin-preload

```diff
@@ -0,0 +1,132 @@
+    via the corresponding environment variables:
+
+    --app          APP_ID
+    --img          IMAGE
+    --api-token    API_TOKEN
+    --api-key      API_KEY
+    --api-host     API_HOST
+    --registry     REGISTRY_HOST
+
+  Example:
+
+    ${info.name} --app 123456 --api-token \"xxxx...\" --img /path/to/resin-os.img
+`
+
+function inspect( value ) {
+  return require('util').inspect( value, {
+    colors: process.stdout.isTTY
+  })
+}
+
+function showError( error, details, report ) {
+
+  let message = !details ?
+    `\n[ERROR] ${error.message}\n` :
+    `\n${inspect(error)}\n`
+
+  if(report) {
+    message += `\nLooks like this might be an issue with ${info.name};\n`
+    message += `Please report it at ${info.bugs.url}\n`
+  }
+
+  process.stderr.write(message)
+
+}
+
+process.on('uncaughtException', (error) => {
+  const code = typeof error.code === 'number' ?
+    (error.code || 1) : 1
+  showError(error, true, true)
+  process.exit(code)
+})
+
+let argv = process.argv.slice(2)
+
+if( argv.includes('--help') || argv.includes('-h') ) {
+  process.stderr.write(USAGE)
+  process.exit(0)
+}
+
+if( argv.includes('--version') || argv.includes('-v') ) {
+  process.stdout.write(info.version + '\n')
+  process.exit(0)
+}
+
+const options = {
+  appId: process.env['APP_ID'],
+  image: process.env['IMAGE'],
+  apiToken: process.env['API_TOKEN'],
+  apiKey: process.env['API_KEY'],
+  apiHost: process.env['API_HOST'],
+  registryHost: process.env['REGISTRY_HOST'],
+}
+
+while( argv.length ) {
+  switch( argv.shift() ) {
+    case '--app': options.appId = argv.shift(); break
+    case '--img': options.image = argv.shift(); break
+    case '--api-token': options.apiToken = argv.shift(); break
+    case '--api-key': options.apiKey = argv.shift(); break
+    case '--api-host': options.apiHost = argv.shift(); break
+    case '--registry': options.registryHost = argv.shift(); break
+  }
+}
+
+// Show usage help if no options have been set
+if( !(options.appId || options.image || options.apiToken || options.apiKey) ) {
+  process.stderr.write(USAGE)
+  process.exit(1)
+}
+
+const preload = require('..')
+
+process.stderr.write('\nBuilding preloader image...\n\n')
+
+preload.build({ stdio: 'inherit' }).once('exit', (code, signal) => {
+
+  if(code !== 0) {
+    const error = new Error(`Preloader build exited with ${code == null ? 'signal' : 'code'} ${code == null ? signal : code}`)
+    error.code = code
+    error.signal = signal
+    showError( error )
+    process.exit( code || 1 )
+  }
+
+  process.stderr.write('\nPreloading...\n\n')
+
+  preload.run(options, { stdio: 'inherit' }).once('exit', (code, signal) => {
+
+    if(code !== 0 || signal != null) {
+      const error = new Error(`Preloader exited with ${code == null ? 'signal' : 'code'} ${code == null ? signal : code}`)
+      error.code = code
+      error.signal = signal
+      showError( error )
+      process.exit( code || 1 )
+    }
+
+    process.stderr.write('\n[OK] Done.\n')
```

Comment:
Yeah, `process.stdout.write` is just a tiny bit faster, which I tend to use in CLIs, as every ms can make it feel sluggish. Changed it to use `console.log()/error()`.